app/components/Panel.js

```diff
@@ -44,25 +43,32 @@ export default class Panel extends Component {
     });
 
     backgroundPageConnection.onMessage.addListener((logItem, sender) => {
-      let mutations = logItem.mutations;
-      let mutationsArray = Object.keys(mutations).map(function(key, index) {
-        return [key, mutations[key]];
-      });
-      // chose 10 arbitrary so we only display 10 mutations in log
-      mutationsArray = mutationsArray.slice(
-        mutationsArray.length - 10,
-        mutationsArray.length
-      );
-      mutations = {};
-      mutationsArray.forEach(function(m) {
-        mutations[m[0]] = m[1];
-      });
-      const slimItem = {
-        state: {
-          mutations: logItem.mutations,
-          queries: logItem.queries
-        }
-      };
+      let slimItem;
+
+      if (logItem.queries) {
+        slimItem = {
+          state: { queries: logItem.queries }
+        };
+      }
+
+      if (logItem.mutations) {
+        let mutations = logItem.mutations;
+        let mutationsArray = Object.keys(mutations).map(function(key, index) {
+          return [key, mutations[key]];
+        });
+        // chose 10 arbitrary so we only display 10 mutations in log
```

Comment:
Don't we already do this in the background now? If not, we should do it there for sure.